proxy/src/cplane_api.rs

```diff
@@ -16,20 +14,21 @@ pub struct DatabaseInfo {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-pub struct ProxyAuthResult {
-    pub ready: bool,
-    pub error: Option<String>,
-    pub conn_info: Option<DatabaseInfo>,
+#[serde(untagged)]
+enum ProxyAuthResponse {
+    Ready { conn_info: DatabaseInfo },
+    Error { error: String },
+    NotReady { ready: bool }, // TODO: get rid of `ready`
```

Comment:
Just curios, am I correct that `serde` tries to match enum variants sequentially and discards all unmatched fields?

Why I ask is that `{ ready: true, conn_info: ... }` should match both `Ready` and `NotReady`, IIUC from the docs. And we protect ourselves from matching it with `NotReady` by the order of arms in the `match` bellow

Reply:
Correct, that's how [untagged](https://serde.rs/enum-representations.html#untagged) mode works. What I'd love to see instead is the proper internally/externally tagged json, but I don't want to change too many things across two repos at the moment. Feel free to introduce such a change after we've merged this.